### PR TITLE
Add transaction and broker fee options

### PR DESCRIPTION
**Pull request by susruth**
Third-party fees can be added to an atomic swap by the third-party application that is requesting the swap. The fee should be specified as an absolute value, and must be displayed in the visual notification approved by the Swapperd user.

Fees are always in addition to the funds being transferred. This must be taken into account by Swapperd when it checks its available balances for the feasibility of a swap.

Fees must be specified (and default to 0). In a delayed swap, the fee is a maximum fee and the actual fee returned by the callback can be less (but not more). This is similar to how the exchange rate approval works for delayed swaps.

### CODE REVIEW

File: adapter/binder/erc20/erc20.go [hunk 1]
```diff
@@ -39,12 +39,14 @@ func NewERC20SwapContractBinder(account beth.Account, swap foundation.Swap, logg
 		return nil, err
 	}
 
+	fmt.Println("Swapperd Address: ", swapperAddress.String())
```
Comment (loongy) [inline adapter/binder/erc20/erc20.go]: Use logger.

Comment (loongy) [reply adapter/binder/erc20/erc20.go]: Not important, just remember for future: pretty sure Logrus has `Infof` etc so you don't need to use `Sprintf` too.

File: adapter/server/handler.go [hunk 2]
```diff
@@ -3,77 +3,106 @@ package server
...
-	return resp
+	return resp, nil
 }
 
-func (handler *handler) GetBalances() (GetBalancesResponse, error) {
-	return handler.fundManager.Balances()
+func (handler *handler) GetBalances(balanceQueries chan<- balance.BalanceQuery) GetBalancesResponse {
+	response := make(chan map[blockchain.TokenName]blockchain.Balance)
+	query := balance.BalanceQuery{
+		Response: response,
```
Comment (loongy) [inline adapter/server/handler.go]: Be consistent with naming. The channel is not the `Response`, it is the `Responder`.

File: adapter/server/http.go [hunk 3]
```diff
@@ -75,6 +84,25 @@ func recoveryHandler(h http.Handler) http.Handler {
 	})
 }
 
+// postBootloadHandler handles the post login request, it loads pending swaps and
+// historical swap receipts into memory.
+func postBootloadHandler(reqHandler Handler, receipts chan<- swap.SwapReceipt, swaps chan<- swap.SwapBlob) http.HandlerFunc {
```
Comment (loongy) [inline adapter/server/http.go]: An interesting idea. I like it, I assume the native application will prompt the use to login to enable the Swapperd after a crash/reboot.

Comment (susruth) [reply adapter/server/http.go]: yes

File: adapter/binder/erc20/erc20.go (hunk 1 above)
Comment (susruth) [reply adapter/binder/erc20/erc20.go]: this was a temporary log, should have been removed.

File: adapter/server/handler.go (hunk 2 above)
Comment (susruth) [reply adapter/server/handler.go]: ok

**Pull request merged**